viewing_party/party.py

```diff
@@ -0,0 +1,243 @@
+    friend_list = user_data['friends']
+
+    #iterating thru each friend's movie dictionary list
+    #set friend_movie_list to hold the dictionary of of movies watched from friends
+    for friend_num in range(len(friend_list)):
+        friend_movie_list = friend_list[friend_num]['watched']
+        #for each movie in friend_movie_list, add movie title to friend_movie_set
+        for movie in friend_movie_list:
+            friend_movie_set.add(movie['title'])
+
+    return friend_movie_set
+
+#returns list of dictionary that friends have seen that user has not seen
+def get_friends_unique_watched(user_data):
+    #initalize variables
+    friend_unqiue_movie_watched = []
+
+    #call set-making functions and capture return in assigned var
+    my_movie_set = make_my_movie_set(user_data) 
+    friend_movie_set = make_friend_movie_set(user_data)
+    #convert difference set to tuple to iterate and make new dictionary
+    movie_titles_unqiue = tuple(friend_movie_set - my_movie_set)
+    
+    #add title to unique_movie_watched list
+    for title in movie_titles_unqiue:
+        friend_unqiue_movie_watched.append({'title': title})
+
+    return friend_unqiue_movie_watched
+
+######################################################
+                # Wave 4 #
+######################################################
+
+#takes user_data and returns a list of reccomended movies referencing
+#friend's movie watched list and user subscription
+def get_available_recs(user_data):
+    #intialize variables
+    rec_list = []
+    movie_title_user = []
+
+    #add movie title to movie_title_user list
+    for movie_user in user_data['watched']:
+        movie_title_user.append(movie_user['title'])
+
+    #access friend key in dict for friend list
+    for friend in user_data['friends']:
+        friend_movies_watched_list = friend['watched']
+
+        #make boolean list from friend watched list
+        for movie_item in friend_movies_watched_list:
+            user_has_seen_movie = movie_item['title'] in movie_title_user
+            user_has_subscription = movie_item['host'] in user_data['subscriptions']
+
+            if not user_has_seen_movie and user_has_subscription and movie_item not in rec_list:
+                rec_list.append(movie_item)
+
+    return rec_list
+
+
+######################################################
+                # Wave 5 #
+######################################################
+
+#make recommendation list of movies dictionaries by genre and
+#friend's watched list
+def get_new_rec_by_genre(user_data):
+    #intialize variables
+    rec_list = []
+
+    #call get_most_watched_grenre function and return 
+    user_most_watched_genre = get_most_watched_genre(user_data)
+
+    #call get_friends_unique_watched and returns dictionary of movie titles that
+    #friends have seen but user has not seen
+    movies_friend_watched = get_friends_unique_watched(user_data)
+    
+    #add genre to movies_friend_watched list and..
```

Comment:
I wasn't sure what this loop was doing, and then I read your comments -- This is a great example of using comments when the code gets a bit complex. 

This is a lot of work just to add the genre information back in. Offhand I'm not sure how you would implement this, but I would consider if there's a way you could refactor your other code so that you didn't lose the genre information from your helper functions.